src/components/gridContainer.js

```diff
@@ -456,7 +462,66 @@ class GridContainer extends Component {
             this.setState({statusMessage: message})
         }
     }
+
+    // validation methods to determine whether the table meets accessibility standards
+    tableHasAtLeastOneHeaderRowOrColumn() {
+        if (this.state.metaHeadercols == 0 && this.state.metaHeaderrows == 0) {
+            this.onError(no_headers_error_message);
+            return false;
+        }
+        return true;
+    }
     
+    tableHasAppropriateHeaderColumnsSet() {
+        const topRow = this.state.handsontableData[0];
+        const numberOfRequiredHeaderColumns = this.calculateNumberOfHeaderColumnsRequired(topRow)
+
+        if (parseInt(this.state.metaHeadercols) < numberOfRequiredHeaderColumns) {
+            this.onError(`Incorrect number of header columns set. Expected at least ${numberOfRequiredHeaderColumns}, got ${this.state.metaHeadercols}`);
+            return false;
+        }
+        return true
+    }
+
+    calculateNumberOfHeaderColumnsRequired(topRow) {
```

Comment:
Can this grab the `topRow` from state? I don't think this needs to be a parameter